addons/pos_coupon/static/src/js/coupon.js

```diff
@@ -1074,5 +1087,20 @@ odoo.define('pos_coupon.pos', function (require) {
             }
             return result;
         },
+        get_full_product_name: function () {
+            if (this.is_program_reward) {
+                var full_product_name = this.product.display_name;
+                var taxes = this.get_taxes()
+                                .map((tax) => tax.name);
+                if (taxes.length) {
+                    taxes = taxes.reduce((acc, val) => acc + ", " + val);
+                    full_product_name += ", applied on products with taxes : " + taxes;
+                } else {
+                    full_product_name += ", applied on products without tax";
+                }
+                return full_product_name;
+            }
+            return _orderline_super.get_full_product_name.apply(this);
+        },
```

Comment:
I'm not sure of this. We need to ask our PO on this.

Reply:
That's quite verbose indeed. I would advise to add the suffix "(tax applied)" when it does.
E.g. "20.0 USD discount on total amount (tax applied)"

For tax details, User can look at the product info.

Reply:
Done. And as we agreed, the discount applied on products without taxes :
> 20.0 USD discount on total amount (no tax)